### DISCUSSION TITLE
API-SERVER is unreachable:<urlopen error [Errno 111] Connection refused>

### DISCUSSION THREAD

**Discussion by ghost** [HewlettPackard/swarm-learning#49] [Q&A]
When I execute MNIST example, meet some problems, and hoping for answers. As shown in the following figure:
![1](https://user-images.githubusercontent.com/62758182/132305313-19759387-fe25-40af-9db3-173307ecff3d.png)
before this, I have downloaded and setup docker images and evaluation licenses.

**Comment by RadhakrishnaJ** [comment]
> When I execute MNIST example, meet some problems, and hoping for answers. As shown in the following figure:
> ![1](https://user-images.githubusercontent.com/62758182/132305313-19759387-fe25-40af-9db3-173307ecff3d.png)
> before this, I have downloaded and setup docker images and evaluation licenses.

Please provide command how you are running SL container? 
Cross check you have apls running and licenses are configured properly.
For MNIST example, sn container should be running before SL run. Cross check you have running sn container, before run-sl.

**Comment by ghost** [comment]
I have noticed what you said and checked the operation procedure. And no problems were found. 
Could you please see what is wrong with my operation steps
In the three images below,I have downloaded and setip docker images and evaluation licenses.
![Snipaste_2021-09-14_09-18-54](https://user-images.githubusercontent.com/62758182/133179469-e5bb152b-0bc3-452a-b0db-e02afcb56aa7.png)
![Snipaste_2021-09-14_09-19-37](https://user-images.githubusercontent.com/62758182/133179473-ac87793e-ae16-47b3-bc54-668392fa9fbe.png)
![Snipaste_2021-09-14_09-20-27](https://user-images.githubusercontent.com/62758182/133179478-57c34d3e-a955-49fd-b5be-2213e3f919a8.png)
Pull all signed images with tag'0.3.0', I already did, but I didn't put up the picture.

In the next picture, I created workspace and initialize Swarm Learning
![Snipaste_2021-09-14_09-18-54](https://user-images.githubusercontent.com/62758182/133179608-82e36db5-a773-4791-af7a-1993dcabf523.png)

Then I think the next skip is run decentralized with different nodes, I open a new terminals to do it, but meet problems
![Snipaste_2021-09-14_09-40-18](https://user-images.githubusercontent.com/62758182/133180587-2c729ac4-f64e-4f19-beb6-6f6b7dd2c392.png)
![Snipaste_2021-09-14_09-40-38](https://user-images.githubusercontent.com/62758182/133180592-95658342-1713-4b8b-aa65-2c05e660b663.png)
![Snipaste_2021-09-14_09-41-31](https://user-images.githubusercontent.com/62758182/133180596-c2eba2d3-3618-401a-b135-c865a0431624.png)

**Reply by RadhakrishnaJ** [reply]
172.18.0.1 is the IP of a system where APLS is running?

**Reply by ghost** [reply]
Thank you for your reply, which provided me with new ideas
At the beginning, I set this IP as the local IP. After listening to your speech, I changed it to the IP of other computers, and I am also running swarm-learning.But I don't know how far the machine corresponding to this IP should go

**Reply by ghost** [reply]
Only run APLS ? That is install Swarm License?

**Reply by ghost** [reply]
![图片](https://user-images.githubusercontent.com/62758182/133433022-84a6daad-e473-4ead-bcbf-1122993d7da2.png)
I seem to have successfully run the instance, thank you very much!!!

**Reply by MZW99** [reply]
I also meet this problem and haven't solved it until now. I did all the steps same as you. Could you please tell me how you solve this problem in the end? Thanks.
